applications/LinearSolversApplication/custom_utilities/mkl_utilities.cpp

```diff
@@ -0,0 +1,76 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+#pragma once
+
+// External includes
+#include <mkl.h>
+
+// Project includes
+#include "custom_utilities/mkl_utilities.h"
+
+namespace Kratos
+{
+
+bool MKLUtilities::CheckThreadNumber(const int NumberOfMKLThreads)
+{
+    const int number_of_threads_mkl = mkl_get_max_threads();
+    if (NumberOfMKLThreads > 0) { // Manual setting
+        if (number_of_threads_mkl != NumberOfMKLThreads) {
+            KRATOS_WARNING("EigenPardisoLUSolver") << "The number of threads in MKL is: " << NumberOfMKLThreads << " instead of " << number_of_threads_mkl << std::endl;

```

Comment:
```suggestion
            KRATOS_WARNING("MKL") << "The number of threads in MKL is: " << NumberOfMKLThreads << " instead of " << number_of_threads_mkl << std::endl;
```